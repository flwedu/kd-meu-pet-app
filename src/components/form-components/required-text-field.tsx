type Props = {
  name: string;
  value: string;
  type: string;
  placeholder: string;
  spanText?: string;
  bordered?: boolean;
  onChange: (e: any) => void;
};

export function RequiredTextField({
  name,
  value,
  type,
  placeholder,
  spanText,
  bordered,
  onChange,
}: Props) {
  return (
    <div className={bordered ? "form-field" : "field"}>
      {spanText && <span>{spanText}</span>}
      <input
        className="input"
        type={type}
        name={name}
        value={value}
        placeholder={placeholder}
        required
        onChange={onChange}
      />
    </div>
  );
}
