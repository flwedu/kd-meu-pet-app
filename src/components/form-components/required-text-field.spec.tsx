import { fireEvent, render } from "@testing-library/react";
import { RequiredTextField } from "./required-text-field";

describe("Required Text Field component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("Should render a text component properly", () => {
    const props = {
      name: "test",
      type: "text",
      placeholder: "Test placeholder",
      bordered: true,
      value: "",
      onChange: jest.fn(),
    };

    const { container } = render(<RequiredTextField {...props} />);

    expect(container).toMatchSnapshot();
  });

  test("Should render a password component properly", () => {
    const props = {
      name: "test",
      type: "password",
      placeholder: "Test placeholder",
      value: "",
      onChange: jest.fn(),
    };

    const { container } = render(<RequiredTextField {...props} />);

    expect(container).toMatchSnapshot();
  });

  test("Should call the onChange method", () => {
    const props = {
      name: "test",
      type: "text",
      placeholder: "Test placeholder",
      value: "",
      onChange: jest.fn(),
    };

    const { container } = render(<RequiredTextField {...props} />);
    const input = container.querySelector("input")!;

    fireEvent.change(input, { target: { value: "123" } });
    expect(props.onChange).toBeCalledTimes(1);
  });
});
