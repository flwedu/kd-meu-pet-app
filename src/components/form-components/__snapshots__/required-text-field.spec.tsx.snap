// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Required Text Field component Should render a password component properly 1`] = `
<div>
  <div
    class="field"
  >
    <input
      class="input"
      name="test"
      placeholder="Test placeholder"
      required=""
      type="password"
      value=""
    />
  </div>
</div>
`;

exports[`Required Text Field component Should render a text component properly 1`] = `
<div>
  <div
    class="form-field"
  >
    <input
      class="input"
      name="test"
      placeholder="Test placeholder"
      required=""
      type="text"
      value=""
    />
  </div>
</div>
`;
